pymeasure/instruments/agilent/agilent4284A.py

```diff
@@ -0,0 +1,288 @@
+        """
+        param_dict = {
+            "frequency": ("FREQ", (20, 1e6)),
+            "voltage": ("VOLT", self._ac_voltage_values),
+            "current": ("CURR", self._ac_current_values),
+            "bias_voltage": ("BIAS:VOLT", self._bias_voltage_values),
+            "bias_current": ("BIAS:CURR", self._bias_current_values)
+        }
+
+        if sweep_mode not in param_dict:
+            log.warning(
+                "Sweep mode but be one of %s, not '%s'.", list(param_dict.keys()), sweep_mode
+            )
+            return
+
+        low_limit = param_dict[sweep_mode][1][0]
+        high_limit = param_dict[sweep_mode][1][1]
+        if (min(sweep_values) < low_limit or max(sweep_values) > high_limit):
+            log.warning(
+                "%s values are outside valid Agilent 4284A range of %g and %g "
+                "and will be truncated.", sweep_mode, low_limit, high_limit
+            )
+            sweep_truncated = []
+            for val in sweep_values:
+                if val <= high_limit and val >= low_limit:
+                    sweep_truncated.append(val)
+            sweep_values = sweep_truncated
+
+        loops = (len(sweep_values) - 1) // 10  # 4284A sweeps 10 points at a time
+        param_div = []
+        for i in range(loops):
+            param_div.append(sweep_values[10*i:10*(i+1)])
+        param_div.append(sweep_values[loops*10:])
+
+        self.clear()
+        self.write("TRIG:SOUR BUS;:DISP:PAGE LIST;:FORM ASC;:LIST:MODE SEQ;:INIT:CONT ON")
+
+        a_data = []
+        b_data = []
+        sweep_return = []
+        for i in range(loops + 1):
+            param_str = ",".join(['%g' % p for p in param_div[i]])
+            self.write(f"LIST:{param_dict[sweep_mode][0]} {param_str};:TRIG:IMM")
+            status_event_register = int(self.ask("STAT:OPER?"))
+            while (status_event_register & 8) != 8:  # Sweep bit no. 3
+                sleep(0.1)
+                status_event_register = int(self.ask("STAT:OPER?"))
+            measured = self.values("FETCH?")
+            # gets 4-ples of numbers, first two are data A and B
+            a_data += [measured[_] for _ in range(0, 4 * len(param_div[i]), 4)]
+            b_data += [measured[_] for _ in range(1, 4 * len(param_div[i]), 4)]
+            sweep_return += self.values(f"LIST:{param_dict[sweep_mode][0]}?")
+
+        # Return to manual trigger and reset display
+        self.write(":TRIG:SOUR HOLD;:DISP:PAGE MEAS")
+        self.check_errors()
+        return a_data, b_data, sweep_return
+
+    def trigger(self):
+        """Execute a bus trigger, which can be used when :meth:`~.trigger_on_bus`
+        is configured. Returns result of triggered measurement.
+        """
+        return self.values("*TRG")
+
+    def trigger_on_bus(self):
+        """Configure the trigger to detect events based on the bus trigger, which can be
+        activated by :meth:`~.trigger`.
+        """
+        self.write("TRIG:SOUR BUS")

```

Comment:
How do you configure it? It seems, that you just enable it.

Reply:
This was something I copied over from another driver without double-checking. I've fixed this and added some trigger-related properties and methods.